lib/sth_database.js

```diff
@@ -1841,6 +1841,490 @@ function getNotificationInfo(data, callback) {
   );
 }
 
+/**
+ * Drops a collection
+ * @param {string} collectionName The name of the collection to drop
+ * @param {string} service The service associated to the collection to drop
+ * @param {function} callback The callback to call with error or the result of the operation
+ */
+function dropCollection(collectionName, service, callback) {
+  db.db(getDatabase(service)).dropCollection(collectionName, callback);
+}
+
+/**
+ * Drops a hashed collection in case the information provided in the data is correct
+ * @param {string} collectionName The hashed name of the collection to drop
+ * @param {object} data It is an object including the following properties:
+ *  - {boolean} aggregated Flag indicating if the collection to drop refers to a aggregated data one or
+ *      not (to a raw data one)
+ *  - {string} service The service associated to the collection to drop
+ *  - {string} servicePath The service path to the collection to drop
+ *  - {string} entityId The entity id to the collection to drop
+ *  - {string} entityType The entity type to the collection to drop
+ *  - {string} attrName The attribute name to the collection to drop
+ * @param {object} options It is an object including the following properties:
+ *  - {boolean} isAggregated Flag indicating if the operation refers to aggregated data. Raw data otherwise
+ * @param callback The callback to call with error or the result of the operation
+ */
+function dropHashedCollection(collectionName, data, options, callback) {
+  var service = data.service,
+      servicePath = data.servicePath,
+      entityId = data.entityId,
+      entityType = data.entityType,
+      attrName = data.attrName;
+  var isAggregated = options.isAggregated;
+
+  getCollection(
+    {
+      service: service,
+      collection: sthConfig.COLLECTION_PREFIX + 'collection_names'
+    },
+    {
+      isAggregated: false,
+      shouldCreate: false,
+      shouldStoreHash: false,
+      shouldTruncate: false
+    },
+    function (err, hashCollection) {
+      if (err) {
+        return process.nextTick(callback.bind(null, err));
+      }
+      hashCollection.findOne(
+        {
+          _id: collectionName
+        },
+        function(err, result) {
+          if (err) {
+            return process.nextTick(callback.bind(null, err));
+          }
+          if (result && result.isAggregated === isAggregated &&
+            result.service === service && result.servicePath === servicePath &&
+            (entityId ? result.entityId === entityId : true) &&
+            (entityType ? result.entityType === entityType : true) &&
+            (attrName ? result.attrName === attrName : true)) {
+            db.db(getDatabase(service)).dropCollection(
+              collectionName,
+              function(err) {
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-sth-comet/commit/787c2bfa1902e97d6934f87c6443edc0b05a17be